### PR TITLE
fix: Popover display empty div when title and content is null

### LINKED ISSUE #42216: Popover 组件当content值为空时，弹出空白
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ji-ben-antd-5-4-7-forked-ghnwme)

### Steps to reproduce

Popover title 跟 content属性均传空值

### What is expected?

不展示气泡卡片

### What is actually happening?

弹出空白卡片

| Environment | Info |
| --- | --- |
| antd | 4.24.10 |
| React | 18 |
| System | Win10 |
| Browser | Chrome112 |

---
4.x与5.x均有次问题，请务必修复下4.x版本

### PR DESCRIPTION
**Pull request by hairgc**
* fix: Popover display empty div when title and content is null

* test: Popover add test

* lint: remove useless block statement

---------

Co-authored-by: MaHui <mahuiyoung@cmbchina.com>
(cherry picked from commit 4271ff0976099619005509de57d24381f9fa34e5)

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #42216
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix: Popover display empty div when title and content is null       |
| 🇨🇳 Chinese |      修复: 当title和content属性均为空值时，Popover组件展示空白气泡     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 36f2518</samp>

Refactored and tested the Popover component. Improved the test cases by using `document.querySelector` and checking the popup element. Simplified the component rendering and avoided empty overlays.

### <samp>🤖 Generated by Copilot at 36f2518</samp>

*  Refactor Overlay component to use a single JSX fragment and render only if title or content props are truthy ([link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0L22-R27), [link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0L62-R62))
*  Reorder import statements in `index.tsx` to follow convention ([link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0L2-R7))
*  Simplify test cases for showing and hiding the overlay in `index.test.tsx` by removing ref prop and using document.querySelector instead of getPopupDomNode ([link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-aba347c569ea59c8e23c5f2dea72628b23a47705e50cd9ea98c578bad2a38184L11-R19), [link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-aba347c569ea59c8e23c5f2dea72628b23a47705e50cd9ea98c578bad2a38184L28-R27), [link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-aba347c569ea59c8e23c5f2dea72628b23a47705e50cd9ea98c578bad2a38184L38-R34), [link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-aba347c569ea59c8e23c5f2dea72628b23a47705e50cd9ea98c578bad2a38184L53-R50), [link](https://github.com/ant-design/ant-design/pull/42229/files?diff=unified&w=0#diff-aba347c569ea59c8e23c5f2dea72628b23a47705e50cd9ea98c578bad2a38184L65-R62))

### CODE REVIEW

**Comment by afc163** [comment]
图片

**Comment by hairgc** [comment]
> 图片

已修复

File: components/popover/__tests__/index.test.tsx [hunk 1]
```diff
@@ -8,34 +8,30 @@ describe('Popover', () => {
...
       </Popover>,
     );
-
-    expect(ref.current.getPopupDomNode()).toBe(null);
-    expect(popover.container.querySelector('.ant-popover-inner-content')).toBeFalsy();
+    expect(document.querySelector('.ant-popover')).toBe(null);
     fireEvent.click(popover.container.querySelector('span')!);
-    expect(popover.container.querySelector('.ant-popover-inner-content')).toBeTruthy();
+    const popup = document.querySelector('.ant-popover')!;
+    expect(popup.querySelector('.ant-popover-inner-content')).toBeTruthy();
```
Comment (afc163) [inline components/popover/__tests__/index.test.tsx:19]:
```suggestion
    expect(popup?.querySelector('.ant-popover-inner-content')).toBeTruthy();
```

是不是改这里更合适。

Comment (hairgc) [reply components/popover/__tests__/index.test.tsx:19]: 嗯嗯，可以的，那我优化下

Comment (hairgc) [reply components/popover/__tests__/index.test.tsx]: 已推

File: components/popover/index.tsx [hunk 2]
```diff
@@ -59,7 +54,12 @@ const Popover = React.forwardRef<unknown, PopoverProps>((props, ref) => {
       {...otherProps}
       prefixCls={prefixCls}
       ref={ref}
-      overlay={_overlay || <Overlay prefixCls={prefixCls} title={title} content={content} />}
+      overlay={
+        _overlay ||
+        (title || content ? (
+          <Overlay prefixCls={prefixCls} title={title} content={content} />
+        ) : null)
+      }
```
Comment (afc163) [inline components/popover/index.tsx:62]: 这里的 || 和 三元 有点难读，提一个 getOverlay 函数吧。

Comment (hairgc) [reply components/popover/index.tsx:62]: 哈哈，那就又回到了最初的版本了，我优化下

Comment (hairgc) [reply components/popover/index.tsx]: 再看下，这里用了useMemo，感觉使用useMemo更合适

**Review by afc163 — APPROVED** [review]

**Pull request merged**